### ISSUE TITLE
No Async support?

### ISSUE THREAD

**Issue by kendallb** [EasyPost/easypost-csharp#66]
This library currently does not have any support for Async network calls? I might modify the code for the library to make it async myself, but I am curious why async support was not added from the start?

**Comment by att14** [comment]
See #42

**Comment by kendallb** [comment]
My new Async library is now complete and I have it as a fork on Github here:

https://github.com/kendallb/easypost-async-csharp

Note that I renamed the project since the changes are quite vast. I re-implemented all the API functions to hang off a single EasyPostClient with a corresponding IEasyPostClient interface so everything is instance based only. This allows me to properly mock all the EasyPost functions in our code and works nicely in a dependency injection environment like ours. 

Since this library is targeting only .Net 4.5 and later I made the entire library Async and did not implement any non-async versions. Any code can use the async functions in a non-async manner easily enough (Unit Tests are written that way). I also took the liberty of adding documentation to everything inline to help me when using Intellisense and consuming the library. 

I also did not like how some the API code was using dictionaries for parameter passing because it makes it much harder to refactor, so all the code was re-implemented to use instance only API's and the dictionaries are kept 100% internal to the API. With that change I also decided to change the naming convention for all the classes to match normal C# style CamelCase rather than the json_case style that is passed over the write. The code to convert a resource object to a dictionary automatically maps all the property names from CamelCase to camel_case, and the RestSharp JSON deserializer takes care of mapping it the other way with the incoming results. 

Finally due to the way exceptions work with async functions I removed the exception support from the code and the functions won't throw exceptions anymore. Rather for anything that is a first class EastPost object that is returned over the wire, if there was an error we deserialize the error object and stuff it into the response returned to the caller. So rather than catching exceptions the caller simply knows they will always get a valid result and can just check for the presence of result.RequestError and if present all the error message details are included in that object. 

I think the API is way cleaner now  and it will suit our purposes nicely. I hope you find it useful and perhaps would consider making it an official version of the API for others looking for C# Async support and/or the ability to use EasyPost in a DI environment?
